src/components/create-safe/steps/OwnerPolicyStep.tsx

```diff
@@ -53,6 +53,11 @@ const OwnerPolicyStep = ({ params, onSubmit, setStep, onBack }: Props): ReactEle
     append({ name: '', address: '' })
   }
 
+  const threshold = watch('threshold')
+  useEffect(() => {
+    setValue('threshold', threshold > fields.length ? fields.length : threshold)
+  }, [fields, setValue, threshold])
```

Comment:
Can't you just validate the select instead?

Reply:
Also, if we populate the select options, how can it even have a threshold bigger than owners?

Reply:
If it doesn't re-render, add a `key={fields.length}` property to the select

Reply:
The form keeps the previously selected threshold.

Reply:
> Can't you just validate the select instead?

But you will have to set the threshold anyway no? My thought here is that if we select a "N out of N" policy that we keep it

Reply:
The last selected threshold could be passed in defaultValue. Math max of that and owners.

Reply:
I was trying it out but didn't quite work as you suggest: setting the value in the `<Select />` does not set the form value unfortunately.

```tsx
<Select
  {...register('threshold')}
  defaultValue={threshold || defaultThreshold}
  value={threshold > fields.length ? fields.length : threshold}
  key={fields.length}
>

Reply:
`setValue` is required because the RHF state has to be updated as well.

Reply:
Yes, because of that I couldn't get around the `useEffect`

Reply:
Let me play with it, I'll get back to you.